projects/vdk-core/src/vdk/internal/builtin_plugins/connection/decoration_cursor.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from typing import Container
+from typing import Optional
+from typing import Tuple
+
+from vdk.internal.builtin_plugins.connection.pep249.interfaces import PEP249Cursor
+
+
+class ManagedOperation:
+    """
+    Query and parameters DTO.
+    """
+
+    def __init__(self, operation: str, parameters: Optional[Container]):
+        self.__operation = operation
+        self.__parameters = parameters
+        self.__operation_decorated = operation
+        self.__parameters_decorated = parameters
+
+    def get_initial_operation(self):
+        return self.__operation
+
+    def get_initial_parameters(self):
+        return self.__parameters
+
+    def get_operation(self):
+        return self.__operation_decorated
+
+    def get_parameters(self):
+        return self.__parameters_decorated
+
+    def get_operation_parameters_tuple(self) -> Tuple[str, Optional[Container]]:
+        return self.__operation_decorated, self.__parameters_decorated
+
+    # maybe track modification history
+    def set_operation(self, decorated_operation):
+        self.__operation_decorated = decorated_operation
+
+    def set_parameters(self, decorated_parameters):
+        self.__parameters_decorated = decorated_parameters
+
+    def set_operation_parameters_tuple(self, decorated_operation, decorated_parameters):
```

Comment:
Usually, operation and parameters are used in combination. It is an useful setter I think,

Reply:
Actually, the ManagedOperation DTO getters/setters were purposefully not documented, due to it is obvious - fields are named pretty explanatory, and it's a simple data transfer object.

I have added the type hints, and added documentation suggested. It might seem a bit heavy for such a simple straight-forward DTO - the effect might be the opposite, regarding User Experience.